extensions/amp-recaptcha-input/0.1/amp-recaptcha-input.js

```diff
@@ -13,3 +13,74 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/**
+ * @fileoverview Async Input Element that uses the
+ * amp-recaptcha-service to dispatch actions, and return
+ * recaptcha tokens
+ */
+
+import {CSS} from '../../../build/amp-recaptcha-input-0.1.css';
+import {Layout} from '../../../src/layout';
+import {
+  installRecaptchaService,
+  recaptchaServiceFor,
+} from './amp-recaptcha-service';
+import {isExperimentOn} from '../../../src/experiments';
+import {setStyles, toggle} from '../../../src/style';
+
+/** @const */
+const TAG = 'amp-recaptcha-input';
+
+export class AmpRecaptchaInput extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!./amp-recaptcha-service.AmpRecaptchaService} */
+    this.recaptchaService_ = recaptchaServiceFor(this.win);
+
+    /** @private {boolean} */
+    this.isExperimentEnabled_ = isExperimentOn(this.win, 'amp-recaptcha-input');
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.NODISPLAY;
+  }
+
+  /** @override */
+  buildCallback() {
+    if (!this.isExperimentEnabled_) {
+      return;
+    }
+
+    return this.mutateElement(() => {
+      toggle(this.element);
+      setStyles(this.element, {
```

Comment:
Can we add a comment here to explain the reason behind applying the styles. And why we use `position: absolute`